// @grammyjs/parse-mode/dist/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isTextEntitiesPayload = exports.isMessageTextEntitiesContent = exports.isMediaEntitiesPayload = exports.isExplanationEntitiesPayload = exports.isCaptionEntitiesResult = exports.isCaptionEntitiesPayload = void 0;
const captionEntitiesMethod = new Set([
    "copyMessage",
    "editMessageCaption",
    "sendAnimation",
    "sendAudio",
    "sendDocument",
    "sendPhoto",
    "sendVideo",
    "sendVoice",
]);
const explanationEntitiesMethod = new Set([
    "sendPoll",
]);
const mediaEntitiesMethod = new Set([
    "editMessageMedia",
    "sendMediaGroup",
]);
const textEntitiesMethod = new Set([
    "editMessageText",
    "sendMessage",
]);
function isCaptionEntitiesPayload(method, _payload) {
    return captionEntitiesMethod.has(method);
}
exports.isCaptionEntitiesPayload = isCaptionEntitiesPayload;
function isExplanationEntitiesPayload(method, _payload) {
    return explanationEntitiesMethod.has(method);
}
exports.isExplanationEntitiesPayload = isExplanationEntitiesPayload;
function isMediaEntitiesPayload(method, _payload) {
    return mediaEntitiesMethod.has(method);
}
exports.isMediaEntitiesPayload = isMediaEntitiesPayload;
function isTextEntitiesPayload(method, _payload) {
    return textEntitiesMethod.has(method);
}
exports.isTextEntitiesPayload = isTextEntitiesPayload;
function isCaptionEntitiesResult(result) {
    return result instanceof Object && "caption" in result;
}
exports.isCaptionEntitiesResult = isCaptionEntitiesResult;
function isMessageTextEntitiesContent(content) {
    return content instanceof Object && "message_text" in content;
}
exports.isMessageTextEntitiesContent = isMessageTextEntitiesContent;
